Added drag chain and upper holder with .scad for it

NeoTheFox opened the pull request.

NeoTheFox committed:
Added drag chain and upper holder with .scad for it

dantman commented:
This seems to connect with a single T-nut and screw. Wouldn't it be better to take advantage of the T-slot extrusion and either make another 90˚ bend to fit 2 nuts (for stability) or make it snap directly onto the extrusion (like how [this filament guide](https://www.thingiverse.com/thing:650425) does)?

NeoTheFox commented:
It is doable, however it would be more plastic. I found one screw sufficient, it never unscrewed on me.

dantman commented:
Another alternative modification. On one piece of mjrice's [smartmount](https://github.com/mjrice/wilson/blob/master/stl/smartmount.stl) you can see a small extra bulge of plastic next to a screw hole. That fits into the gap in the middle of the 2020 extrusion and stabilizes the piece preventing it from rotating. That's another possibility here. Although that might make it harder to print flat.

dantman commented:
Could you upload a photo of this on a printer? What extruder head are you using? How did you orient your wires when constructing it?

I started printing this out to try on my printer. It wasn't oriented the way I expected (though my expectations of what direction it would face were probably wrong). But more importantly, I found the wl/holder didn't seem to extend far enough. The chain was at least 2cm (in the Y) away from where my wires came up off my extruder.

NeoTheFox commented:
[Take a look at photos](http://www.thingiverse.com/thing:400027)
This was intended as a part of my [remix](https://github.com/NeoTheFox/wilson-bowden-addon/tree/master/stl), so you can find links to originals and files there.

dantman commented:
Ah, so it's supposed to be on the left. I thought it would go on the middle or right to avoid interfering with any filament being guided into the extruder.

So this profile of mount is designed for your bowden extruder.
On the Wilson's default direct drive extruder, assuming you orient your extruder motor so the wires face upwards (as IIRC mjrice suggests in his assembly video) the location of the bundle of wires is another 2cm away from the frame. 
![img_20150714_045605](https://cloud.githubusercontent.com/assets/53399/8672699/b97dd41c-29e5-11e5-96ab-62b194ba3733.jpg)
![img_20150714_045628](https://cloud.githubusercontent.com/assets/53399/8672698/b97d9b14-29e5-11e5-957f-db97745eacc3.jpg)
((Maybe I got a bad angle of this))

Additionally I have a mild concern about how the chain is in-line with the threaded rod. So if the chain is mounted differently, given enough extra links, or the head moves into just the right spot, the chain may hit the right z-axis threaded rod instead of bumping a non-moving part or clearing the printer entirely. 
![img_20150714_045654](https://cloud.githubusercontent.com/assets/53399/8672700/b9805aca-29e5-11e5-8d03-e9d3f09c3401.jpg)
((I don't have enough links yet, I'm just showing that it is in line with the rods))

dantman committed:
Update the chain holder

- Fix .scad so it actually compiles
- Fix the name of the scad file
- Add 0.4mm to the screwhole diameter (I found it a little tight)
- Redesign chain holder to use variables and be customizable

Redesigned scad supports these customizations:
- Variable bar length (based on a parameter indicating how many mm the wires are offset from the 2020 extrusion).
- Screw padding (so slightly longer screws can secure the holder)
- Multiple optional screwhole locations, 1 inside and 2 outside

Added a chain holder with settings that should work for the Wilson's direct drive extruder (assuming you orient the wires upwards):
- 56mm wire offset
- 2mm screw padding (most of the wilson parts have 4mm+ between the screw head and the T-slot)
- 2 screw holes on the outside (should secure the longer bar better)

dantman committed:
Allow the end facing the front of the printer to be closed.

This can add support and in some cases fix issues printing this part some printers (i.e. my printer) can have.

dantman commented:
I've forked this tree and made some updates to it.

https://github.com/NeoTheFox/wilson/compare/master...dantman:chain
- The new .scad file is parametric and supports changing the length of the bar.
- I added support for padding for screws (the floor is 2mm but the Wilson printed parts and screws in the BOM are made for at least 4mm floors)
- Allowed the location and number of screw holes to be customized (the inner screw hole and 2 outer screw holes can individually be enabled or disabled).

NeoTheFox committed:
Merge pull request #1 from dantman/chain

Chain

NeoTheFox commented:
Wow, this is very cool, thanks!
